controllers/machinepools.go

```diff
@@ -74,8 +74,13 @@ func (r *AppWrapperReconciler) scaleMachinePool(ctx context.Context, aw *arbv1.A
 			m[aw.Name] = aw.Name
 			klog.Infof("The instanceRequired array: %v", userRequestedInstanceType)
 
+			machinePoolTaint := cmv1.NewTaint().
+				Key(aw.Name).
+				Value("value1").
+				Effect("PreferNoSchedule")
```

Comment:
Do we wish to use for the taints effect `PreferNoSchedule` as opposed to `NoSchedule`? The first will still allow other pods without the toleration to be assigned to that node. I think `NoSchedule` would best suit our use case.